osu.Game.Rulesets.Osu/Objects/Slider.cs

```diff
@@ -19,14 +19,14 @@ namespace osu.Game.Rulesets.Osu.Objects
 {
     public class Slider : OsuHitObject, IHasCurve
     {
-        public double EndTime
+        public double EndTime => StartTime + this.SpanCount() * Path.Distance / Velocity;
+
+        public double Duration
         {
-            get => StartTime + this.SpanCount() * Path.Distance / Velocity;
+            get => EndTime - StartTime;
```

Comment:
a little inconsistent maybe that pretty much everything else was adjusted (`Duration` is the primary thing calculated in other places and `EndTime` is its derivative, and here it's the other way around), but it's not very significant

Reply:
Yeah, I think I mentioned in discord that a better direction here would be to write out the velocity to a private variable whenever any of the components affecting it change. Is that the direction you're thinking?

It needs to be done in order to have an interface-level `DurationBindable`, supported in `DrawableHitObject` just like `StartTime` is, which is one of the future goals I had. Just haven't done that yet as we were also discussing whether having bindable property counterparts in `HitObject`s was the correct direction.

Reply:
Right, agreed. For the bindable to happen, internal recomputation of the duration needs to happen any time anything changes that would impact it.

I think there's a good chance this can be relatively self-contained (imagining something like `updateDuration()` inside `Slider` that would recompute this and raise value changed events) but if it pops up/starts being ridiculous in other places then a per-HO invalidation might be the way.